packages/googledrive/src/Adaptor.js

```diff
@@ -0,0 +1,257 @@
+import fs from 'fs';
+import { logError } from './Utils';
+import {
+  execute as commonExecute,
+  composeNextState,
+} from '@openfn/language-common';
+
+import {
+  normalizeOauthConfig,
+  expandReferences,
+} from '@openfn/language-common/util';
+
+import { google } from 'googleapis';
+
+let client = undefined;
+
+/**
+ * @private
+ * Creates a connection to Google Drive using OAuth2 authentication.
+ * @param {Object} state - object containing the access token.
+ * @returns {Object} state with Google Drive client initialized.
+ */
+function createConnection(state) {
+  const { accessToken } = state.configuration;
+  const auth = new google.auth.OAuth2();
+  auth.credentials = { access_token: accessToken };
+  client = google.drive({ version: 'v3', auth });
+  return state;
+}
+
+/**
+ * @private
+ * Removes the Google Drive client connection.
+ * @param {Object} state - object.
+ * @returns {Object} state with client reset.
+ */
+function removeConnection(state) {
+  client = undefined;
+  return state;
+}
+
+/**
+ * Execute a sequence of oper.
+ * Wraps `language-common/execute`, and prepends initial state for http.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(
+      createConnection,
+      ...operations,
+      removeConnection,
+    )({
+      ...initialState,
+      ...state,
+      configuration: normalizeOauthConfig(state.configuration),
+    });
+  };
+}
+
+/**
+ * Uploads a file to Google Drive with resumable upload support.
+ * @public
+ * @example
+ * create({
+ *   filePath: '/path/to/file.txt',
```

Comment:
So OpenFn does not support access to file system, We usually pass file as base64 encode to the destination system. I will have a sync with Joe to brainstorm a bit more around this